docs/userguides/contracts.md

```diff
@@ -21,6 +21,43 @@ dev = accounts.load("dev")
 contract = project.MyContract.deploy(sender=dev)
 ```
 
+You can alternatively use this syntax instead:
+
+```python
+from ape import accounts, project
+
+dev = accounts.load("dev")
+contract = dev.deploy(project.MyContract)
+```
+
+If your contract requires constructor arguments then you will need to pass them to the contract in the [kwargs](../methoddocs/transactions.html) when deploying like this:
+
+```python
+from ape import accounts, project
+
+dev = accounts.load("dev")
+contract = project.MyContract.deploy("argument1", "argument2", sender=dev)
+```
+
+you can alternatively use this syntax instead:
+
+```python
+from ape import accounts, project
+
+dev = accounts.load("dev")
+contract = accounts.dev.deploy(project.MyContract, "argument1", "argument2")
+```
+
+With this technique, you can feed as many constructor arguments as your contract constructor requires.
+
+If you do not pass the correct amount of constructor arguments when deploying, you will get an error showing your arguments don't match what is in the ABI:
+
+```bash
+ArgumentsLengthError: The number of the given arguments (0) do not match what is defined in the ABI (2).
+```
+
+In this case it is saying that you have fed 0 constructor arguments but the contract requires 2 constructor arguments to deploy.
+
```

Comment:
[This](https://github.com/ApeWorX/ape/blob/94e1f7eec8b819716d13c5101153c9b586af62d2/src/ape/contracts/base.py#L1055) hasn't been released yet, but neither have these docs so we can do this to show what the args are that the constructor takes:

```
>>> project.MyContract.constructor
constructor(string argument1, string argument2)
```